### PR TITLE
[Gobblin-1422] Publish to artifactory instead of bintray

### PR DESCRIPTION
**Pull request by Will-Lo**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1422

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):

Based on Bintray's sunset https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/ publishing to bintray can no longer be done by May 1st. We will need to publish jars to an alternative source, such as JFrog's Artifactory.
This PR changes the publishing location to artifactory instead of bintray.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gradle/scripts/artifactoryPublishing.gradle [hunk 1]
```diff
@@ -104,45 +105,22 @@ subprojects{
...
-        mavenCentralSync {
-          sync = false
-        }
+  artifactory {
+    contextUrl = 'https://linkedin.jfrog.io/artifactory/'
+    publish {
+      repository {
+        repoKey = 'gobblin-github'
+        username = project.hasProperty('artifactoryUser') ? project.property('artifactoryUser') : System.getenv('ARTIFACTORY_USER')
+        password = project.hasProperty('artifactoryApiKey') ? project.property('artifactoryApiKey') : System.getenv('ARTIFACTORY_API_KEY')
```
[the remark this answers is not in the document]
Comment (Will-Lo) [reply gradle/scripts/artifactoryPublishing.gradle:114]: Right now it's my personal account

File: .travis.yml [hunk 2]
```diff
@@ -60,6 +60,7 @@ env:
   global:
     - secure: U72nmzXq7kXcIabiwvrTF+WkNFQxov2ACd8oPxWSHsyzRsfVJN42vT7gS3dLbH5G5claXG2p+rur4ueVffzYSwJ8B9OP6gTB8sNJnGr9zyxni4OJchyKqOYuj+UBpEQC/7qXKMCKnuJndsf1OvndDh/V1SH0DSSUuA6mDtgO/eM=
     - secure: WiK7tyFV68xdkIfUlWreUHgEGGjaCBd73O4SbjE9AsbqqF7D+Iu8iRo1OhKQj+6eajUH9Eoev9rVN74FQgUfeNzrOkYsgDysXmyZ7+UxFokijFcATJmIBompA3dySGU2qXeKbJMNuUjXgrRIludaV6h2ahL6Fji42cgK4I3s2qs=
```
Comment (vikrambohra) [inline .travis.yml:62]: The first 2 look like old bintray credentials. Shouldn't we be removing them?

**Review by vikrambohra — APPROVED** [review]
+1 LGTM 
Thank you for the effort. Please have one of the commitors merge it since I am not a committor.

**Review by sv2000 — APPROVED** [review]
+1.

**Pull request closed without merging**